src/azure-cli/azure/cli/command_modules/sqlvm/_params.py

```diff
@@ -158,8 +159,8 @@ def load_arguments(self, _):
                    arg_type=get_location_type(self.cli_ctx),
                    validator=get_default_location_from_resource_group)
         c.argument('expand',
-                   help='Get the SQLIaaSExtension configuration settings.',
-                   arg_type=get_enum_type(['*']))
+                   help='Get the SQLIaaSExtension configuration settings. To view all settings, use *. To select only a few, the settings must be comma-separted. Allowed values: *, AutoBackupSettings, AutoPatchingSettings, KeyVaultCredentialSettings, ServerConfigurationsManagementSettings',
+                   validator=validate_expand)
```

Comment:
please show all enum values here.

Reply:
The SQL VM RP currently does not have an enum for the SQLIaaSExtension configuration settings. I'm using the validate_expand to compare the user input and allowed values. In the help flag, I added manually added the allowed values to communicate with users which values are allowed.

If I were to implement this with enums, I would need to make changes to the SQL VM RP and update the Python model. Would I need to take these steps? Thanks!

Reply:
@Juliehzl Is there a command in CLI that supports a combination of allowed values separated by a comma?

In this expand flag, we want users to be able to input one or more values. It seemed that `arg_type` usually only accepts one value. In `validate_expand`, I check the cases when more than one value is inputted.

Reply:
Example
https://github.com/Azure/azure-cli/blob/f67024b48bd0efea4084f028897d7b7a7abf35c9/src/azure-cli/azure/cli/command_modules/storage/_params.py#L247

Reply:
discussed online. get_enum_type() is not suitable for the case.
We have another typical design for such case. We add nargs='+' with arg_type=get_enum_type([a,b,c,d]). And then customer could use `--expand a b ` to input [a,b] for expand variable.

Reply:
Changes using nargs='+' have been committed.